### DISCUSSION TITLE
Looking for a 3D Plot library

### DISCUSSION THREAD

**Discussion by alessandrofrancesconi** [AvaloniaUI/Avalonia#11731] [Q&A]
Hello everyone, I'm looking for a way to plot this type of 3D surface:

![image](https://github.com/AvaloniaUI/Avalonia/assets/1595223/e82edd6c-589d-48d5-b822-b496482ba0c4)

The library will be integrated in a project based on Avalonia 11 that runs on both desktop and Android/iOS environments (so a [cross platform solution](https://docs.avaloniaui.net/tutorials/developing-for-mobile/create-a-cross-platform-solution)).

Mouse/touch controls are optional.

Can someone suggest something? Thanks!

**Comment by jp2masa** [comment]
I'm working on a 3D plotting library, but it's far from finished (will probably take a few months, it's a hobby). I'm also not sure if I will make it open source yet (maybe just part of it).

**Comment by alessandrofrancesconi** [comment]
Just wondering if there are some updates about this request, can someone help?

**Comment by MikeCodesDotNET** [comment]
Have you had a look at the [Steema charts for Avalonia](https://www.steema.com/product/net_avalonia)?

**Reply by XDgl233** [reply]
Can confirm it does support 3d plotting, though it's not a free solution.
![image](https://github.com/AvaloniaUI/Avalonia/assets/70514198/668cc825-b5e7-4500-8532-de561b573cb3)

**Reply by alessandrofrancesconi** [reply]
Did you test it on mobile?
